sdks/java/io/kafka/upgrade/src/main/java/org/apache/beam/sdk/io/kafka/upgrade/KafkaIOTranslation.java

```diff
@@ -102,6 +102,7 @@ static class KafkaIOReadWithMetadataTranslator implements TransformPayloadTransl
             .addNullableByteArrayField("key_deserializer_provider")
             .addNullableByteArrayField("value_deserializer_provider")
             .addNullableByteArrayField("check_stop_reading_fn")
+            .addNullableLogicalTypeField("consumer_polling_timeout", new NanosDuration())
```

Comment:
You have to also update the `toConfigRow` and `fromConfigRow` methods of the `KafkaIOReadWithMetadataTranslator` to use this new field when generating a PTransform proto from the PTransform object and vice versa.

Reply:
You can probably just copy what we do for `max_read_time` above.

Also, since this is a new property for the current Beam version. We have to not use this field for older Beam versions and set appropriate defaults. For example see here: https://github.com/apache/beam/blob/1e89f834c8438b3c9b20e0de4aa7f5d126a1e172/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIOTranslation.java#L326

Reply:
@chamikaramj The new config consumerPollingTimeout already has a default of 2 seconds. Is the logic similar to [beam/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIOTranslation.java](https://github.com/apache/beam/blob/1e89f834c8438b3c9b20e0de4aa7f5d126a1e172/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIOTranslation.java#L326) still necessary?

Reply:
Yeah, I think if you try to pull an non-existent field from the row, it would fail instead of returning null.

Reply:
got it